src/ExpressiveInstaller/Resources/src/ExpressiveSymfonyContainer.php

```diff
@@ -0,0 +1,192 @@
+            isset($this->config[static::ENABLE_CACHE]) &&
+            $this->config[static::ENABLE_CACHE];
+
+        if (! $cacheEnabled) {
+            $this->cacheContainer($this->build());
+        }
+
+        $container = $this->loadFromCache($this->cachedContainerFile);
+
+        // Inject config
+        $container->set('config', $this->config);
+
+        // Inject services
+        if (! empty($this->config['dependencies']['services'])
+            && is_array($this->config['dependencies']['services'])
+        ) {
+            foreach ($this->config['dependencies']['services'] as $name => $service) {
+                $container->set($name, $service);
+            }
+        }
+
+        return $container;
+    }
+
+    /**
+     * @var string $cachedContainerFile
+     * @return \ProjectServiceContainer
+     */
+    private function loadFromCache($cachedContainerFile)
+    {
+        require_once $cachedContainerFile;
+
+        return new \ProjectServiceContainer();
+    }
+
+    /**
+     * @return ContainerBuilder
+     */
+    private function build()
+    {
+        // Build container
+        $container = new ContainerBuilder();
+
+        // Inject delegator factories
+        // This is done early because Symfony Dependency Injection does not allow modification of a
+        // service after creation. As such, we need to create custom factories
+        // for each service with delegators.
+        if (! empty($this->config['dependencies']['delegators'])
+            && is_array($this->config['dependencies']['delegators'])
+        ) {
+            $this->marshalDelegators($container);
+        }
+
+        // Inject factories
+        foreach ($this->config['dependencies']['factories'] as $name => $object) {
+            $container->register($name)
+                ->setClass($object)
+                ->addArgument(new Reference('service_container'))
+                ->addArgument($object)
+                ->addArgument($name)
+                ->setFactory([\App\CallableFactory::class, 'build']);
+        }
+
+        // Inject invokables
+        foreach ($this->config['dependencies']['invokables'] as $name => $object) {
+            $container->register($name, $object);
+        }
+
+        // Inject aliases
+        foreach ($this->config['dependencies']['aliases'] as $alias => $target) {
+            $container->setAlias($alias, $target);
+        }
+
+        $container->compile();
+
+        return $container;
+    }
+
+    private function marshalDelegators(ContainerBuilder $container)
+    {
+        foreach ($this->config['dependencies']['delegators'] as $service => $delegatorNames) {
+            $factory = null;
+
+            if (isset($this->config['dependencies']['services'][$service])) {
+                // Marshal from service
+                $instance = $this->config['dependencies']['services'][$service];
+                $factory = function () use ($instance) {
```

Comment:
using a closure is not compatible with dumping.

Instead of using your ExpressiveSymfonyDelegatorFactory passing it a closure, your should let the service be registered normally (so remove the code here which unsets the config), and use Symfony's service decoration feature to apply delegators.